plugins/org.eclipse.n4js/src/org/eclipse/n4js/internal/lsp/N4JSProjectConfig.java

```diff
@@ -120,8 +120,18 @@ public IWorkspaceConfig getWorkspaceConfig() {
 		return workspace;
 	}
 
-	/** @return true iff this project should be indexed only */
+	/**
+	 * Projects are indexed only if they either have '.' as the output path or if they are located in node_modules
```

Comment:
Misleading wording: sounds as if projects with an output path other than '.' and not located in a node_modules folder won't be indexed.

Also the "either ... or" is incorrect (the method returns true also for projects that have . as output path AND are located in node_modules folders).